app/api/ltc.py

```diff
@@ -1,11 +1,117 @@
+
+    # drop any rows with empty dates
+    df.drop(df[pd.isnull(df['Date'])].index, inplace = True)
+    df['Date'] = df['Date'].astype(int)
+
+
+def make_matching_column_name_map(df):
+    # make a map of corresponding column names (cumulative to current)
+    num_numeric_cols = 12  # number of metrics
+    first_metric_col = 14  # position of 1st metric, "Cumulative Resident Positives"
+    col_map = {}
+    for i in range(num_numeric_cols):
+        cumulative_col = df.columns[first_metric_col+i]
+        outbreak_col = df.columns[first_metric_col+i+num_numeric_cols]
+        col_map[cumulative_col] = outbreak_col
+    return col_map
+
+
+# takes a dataframe containing the same facility name/date data and collapses the rows.
+# Finds conceptually paired columns based on the content of col_map.
+def collapse_rows_new_header_names(df_group, col_map):
+    if df_group.shape[0] == 1:
+        return df_group
+
+    new_df_subset = df_group.loc[df_group['Outbrk_Status'] == 'OPEN'].copy()
+    row_descriptor = '%s %s %s %s' % (
+        set(new_df_subset['Facility']),
+        set(new_df_subset['State_Facility_Type']),
+        set(new_df_subset['County']), 
+        set(new_df_subset['Date']))
+
+    # expecting only one row/open outbreak; if this isn't true, check that the columns are the same
+    if new_df_subset.shape[0] > 1:
+        deduped = new_df_subset.drop_duplicates()
+        if deduped.shape[0] > 1:
+            print('Multiple open outbreak rows with different data: %s' % row_descriptor)
+        else:
+            print('Duplicate open outbreak rows: %s' % row_descriptor)
+            
+        return df_group
+
+    if new_df_subset.empty:  # no open outbreaks, but we may want to merge some closed ones
+        new_df_subset = df_group.head(1)
+        
+    for colname in col_map.keys():
+        try:
+            cumulative_val = df_group[colname].fillna(0).astype(int).sum()
+            current_open_val = df_group[col_map[colname]].fillna(0).astype(int).sum()
+            val = cumulative_val + current_open_val
+            if val > 0:
+                index = list(df_group.columns).index(colname)
+                new_df_subset.iat[0,index] = val
+        except ValueError:  # some date fields in numeric places, return group without collapsing
+            print('Some non-numeric fields in numeric places, column %s: %s' % (
+                colname, row_descriptor))
+            return df_group
+
+    return new_df_subset
+
+
+def preclean_FL(df):
+    # these aren't real data rows: dropping 
+    df.drop(df[df.County.isin(['TOTAL ICF', 'TOTAL ALF', 'TOTALS'])].index, inplace = True)
+    def process_county(county):
+        if county in ['UNKNOWN', 'UNKNOWN COUNTY']:
+            return ''
+        elif county in ['DADE', 'MIAMI-DADE', 'MIAMIͲDADE', 'MIAMI‐DADE']:
+            return 'MIAMI-DADE'
+        else:
+            return county
+        
+    df['County'] = df['County'].apply(process_county)
+
+
+def process_df(df):
```

Comment:
Can this function get a more descriptive name? It's fine for now, but it might get concerning once we add other types of processing tasks